remixer_core/src/main/java/com/google/android/libraries/remixer/Remixer.java

```diff
@@ -132,6 +135,34 @@ public void addItem(RemixerItem remixerItem) {
     return list;
   }
 
+  /**
+   * Notifies all other instances of variables with the same key that the value has changed and
+   * informs the synchronization mechanism as well.
+   */
+  void onValueChanged(Variable variable) {
+    synchronizationMechanism.onValueChanged(variable);
+    List<RemixerItem> itemList = getItemsWithKey(variable.getKey());
+    for (RemixerItem item : itemList) {
+      if (item != variable) {
+        ((Variable) item).setValueWithoutNotifyingOthers(variable.getSelectedValue());
```

Comment:
It could, I wanted to avoid it though, since that means we need to make sure every implementation of SynchronizationMechanism has to do it and we may need to copy code around.

However that may be the more correct way to deal with this and a proper inheritance tree between them may aid with this. It also ties the SynchronizationMechanism to private methods of Variable and Trigger, I was thinking of a way to avoid using setValueWithoutNotifyingOthers at all from the SynchronizationMechanism

What do you think? do you prefer to move this to the SynchronizationMechanism or to keep it here and then make those internal methods package-local to close access as much as possible?

I actually find it hard to decide between those two and would like your input.